### PR TITLE
Issues 1823

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
This PR is for #1823 

## How Has This Been Tested?
Tested with PHP new standard.

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by raftaar1191** [comment]
@ravinderk 

Please verify the PR for issues https://github.com/WordImpress/Give/issues/1823

**Comment by DevinWalker** [comment]
@raftaar1191 this PR looks like it contains work from #1843 as well as code from other recent PRs. Can you revise it to only show the code from the original PR?

**Comment by raftaar1191** [comment]
@DevinWalker  Done
Remove the #1843 code from the PR

File: includes/admin/payments/class-payments-table.php [hunk 1]
```diff
@@ -227,56 +227,79 @@ public function search_box( $text, $input_id ) {
...
+			 *
+			 * Filter can be used to show all the status inside the donation submenu tabs return true to show all the tab.
+			 *
+			 * @since 1.8.12
+			 *
+			 * @param string  $key Current view tab value.
+			 * @param int  $count Number of donation inside the tab.
+			 */
+			if ( 'all' === $key || $key === $current || apply_filters( 'give_payments_table_show_all_status', 0 < $count, $key, $count ) ) {
+				$count_html = '&nbsp;<span class="count">(' . $count . ')</span>';
```
Comment (ravinderk) [inline includes/admin/payments/class-payments-table.php]: @raftaar1191 can we use `sprintf`? It is for improve code formatting.

File: includes/admin/payments/class-payments-table.php [hunk 2]
```diff
@@ -227,56 +227,79 @@ public function search_box( $text, $input_id ) {
...
+					$href = esc_url( add_query_arg( array( 'status' => $key, 'paged' => false ) ) );
+				}
+
+				if ( ( 'all' === $key && empty( $current ) ) ) {
+					$class = 'class="current"';
+                } elseif ( $current == $key ) {
+					$class = 'class="current"';
+                }
+
+				$views[ $key ] = '<a href="'. $href .'" '. $class .' >'. $tab[1] . $count_html .'</a>';
```
Comment (ravinderk) [inline includes/admin/payments/class-payments-table.php]: @raftaar1191 can we use `sprintf`? It is for improve code formatting.

File: includes/admin/payments/class-payments-table.php (hunk 1 above)
Comment (raftaar1191) [inline includes/admin/payments/class-payments-table.php]: Done

File: includes/admin/payments/class-payments-table.php (hunk 2 above)
Comment (raftaar1191) [inline includes/admin/payments/class-payments-table.php]: Done

**Comment by ravinderk** [comment]
@raftaar1191 Can you update this with `release/1.8.12`

**Comment by raftaar1191** [comment]
@ravinderk 

Done

File: includes/admin/payments/class-payments-table.php [hunk 3]
```diff
@@ -240,56 +240,79 @@ public function search_box( $text, $input_id ) {
...
+			 * Filter can be used to show all the status inside the donation tabs.
+			 *
+			 * Filter can be used to show all the status inside the donation submenu tabs return true to show all the tab.
+			 *
+			 * @since 1.8.12
+			 *
+			 * @param string  $key Current view tab value.
+			 * @param int  $count Number of donation inside the tab.
+			 */
+			if ( 'all' === $key || $key === $current || apply_filters( 'give_payments_table_show_all_status', 0 < $count, $key, $count ) ) {
```
Comment (ravinderk) [inline includes/admin/payments/class-payments-table.php]: @raftaar1191 simplify this code as we discussed

**Comment by ravinderk** [comment]
@raftaar1191 I can see formatting issue in your code. Can you improve formatting.

image

File: includes/admin/payments/class-payments-table.php (hunk 3 above)
Comment (raftaar1191) [inline includes/admin/payments/class-payments-table.php]: Done

**Pull request merged**